pkg/telemetryquerylanguage/functions/tqlotel/README.md

```diff
@@ -130,14 +133,17 @@ Examples:
 
 ## replace_all_patterns
 
-`replace_all_patterns(target, regex, replacement)`
+`replace_all_patterns(target, mode, regex, replacement)`
 
-The `replace_all_patterns` function replaces any segments in a string value that match the regex pattern with the replacement string.
+The `replace_all_patterns` function replaces any segments in a string value or key that match the regex pattern with the replacement string.
+
+`mode` determines whether try to match regex with string value or key. For mode `key`, the processor matches regex with key; for mode `value`, the processor matches regex with value. 
```

Comment:
```suggestion
`mode` determines whether the match and replace will occur on the map's value or key. Valid values are `key` and `value`.
```

Can you also move this entry after line entry for target, regex, and replacement (after line 142)?